File: CS/SaveComponents/Form1.cs
using System;
using System.ComponentModel;
using System.Data;
using System.Windows.Forms;
using System.Collections;
using DevExpress.XtraReports.UI;

namespace SaveComponents {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

    private void Form1_Load(object sender, EventArgs e) {
        XtraReport1 report = new XtraReport1();
        report.SaveComponents += new 
            EventHandler<SaveComponentsEventArgs>(report_SaveComponents);
        report.ShowDesignerDialog();
    }

    void report_SaveComponents(object sender, SaveComponentsEventArgs e) {
        int tableAdapterIdx = GetItemIndex(e.Components, typeof(Component));
        if (tableAdapterIdx >= 0)
            e.Components.RemoveAt(tableAdapterIdx);
        int dsIdx = GetItemIndex(e.Components, typeof(DataSet));
        if (dsIdx >= 0)
            e.Components.RemoveAt(dsIdx);
    }

    private static int GetItemIndex(IList components, Type targetType) {
        int idx = -1;
        for (int i = 0; i < components.Count; i++) {
            if (components[i].GetType().BaseType == targetType) {
                idx = i;
                break;
            } 
        }

        return idx;
    }
    }
}

File: VB/SaveComponents/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Windows.Forms
Imports System.Collections
Imports DevExpress.XtraReports.UI
' ...

Namespace SaveComponents
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) _
        Handles MyBase.Load
            Dim report As New XtraReport1()
            AddHandler report.SaveComponents, AddressOf report_SaveComponents
            report.ShowDesignerDialog()
        End Sub

        Private Sub report_SaveComponents(ByVal sender As Object, _
        ByVal e As SaveComponentsEventArgs)

            Dim tableAdapterIdx As Integer = GetItemIndex(e.Components, _
            GetType(Component))
            If tableAdapterIdx >= 0 Then
                e.Components.RemoveAt(tableAdapterIdx)
            End If
            Dim dsIdx As Integer = GetItemIndex(e.Components, GetType(DataSet))
            If dsIdx >= 0 Then
                e.Components.RemoveAt(dsIdx)
            End If

        End Sub

        Private Shared Function GetItemIndex(ByVal components As IList, _
        ByVal targetType As Type) As Integer

            Dim idx As Integer = -1
            For i As Integer = 0 To components.Count - 1
                If components(i).GetType().BaseType Is targetType Then
                    idx = i
                    Exit For
                End If
            Next i

            Return idx
        End Function
    End Class

End Namespace